### ISSUE TITLE
[HOLD - App  #26239] [$500] Mobile-Safari Auto Focuses Composer in Report on Existing chat

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#29509]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.83-5
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @jeet-dhandha 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1697136130670869

## Action Performed:
1. Open any Chat (which has some messages) in Mobile Safari.
2. Verify Composer is Auto-focused.
3. Test on iOS, Android, Mobile-Chrome for `Step 1`,  Verify that it doesn't auto-focus on chat with messages.
## Expected Result:
 Composer in Mobile safari should not get auto-focused.
## Actual Result:
 Composer in Mobile-Safari get's auto-focused.
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/38435837/4f2adba0-906f-4bef-a516-c1ad953a1950

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/38435837/5ef45fb5-b0e4-407f-869b-057db1453cb1

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/38435837/c37f4502-a2fa-4375-91c0-1d9e9c42b255

https://github.com/Expensify/App/assets/38435837/e8f869c9-f8d3-41da-991a-0fc0052c9bf5

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d1e885de54d568ec</li>
        <li>Upwork Job ID: 1712590372450942976</li>
        <li>Last Price Increase: 2023-10-12</li>
    </ul>
</details>

**Comment by graylewis** [comment]
I'm unable to replicate this on my iOS simulator

**Comment by jeet-dhandha** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- Currently mobile safari is focusing on Composer even if the `shouldAutoFocus` flag is false. Other mobile devices such as iOS, Android and mWeb Chrome does not have this issue. 
- We aren't focusing composer as user might just want to read the existing conversation.

### What is the root cause of that problem?
- For mWeb-Safari we have below code which handles the focus issue.

https://github.com/Expensify/App/blob/bbbfc9cc7acc74d3fb885f409d9610edfcdd5d91/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js#L40-L44

https://github.com/Expensify/App/blob/bbbfc9cc7acc74d3fb885f409d9610edfcdd5d91/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js#L126-L129

- But ComposerWithSuggestion also uses a Component called `Composer` whose `index.js` also has `selection` state which is causing the issue.

https://github.com/Expensify/App/blob/bbbfc9cc7acc74d3fb885f409d9610edfcdd5d91/src/components/Composer/index.js#L175-L178

### What changes do you think we should make in order to solve the problem?
- If we update `Composer` with the code mentioned above in `ComposerWithSuggestion` then it will fix the issue.
- And instead of `shouldAutoFocus` we will use `autoFocus` prop.

### What alternative solutions did you explore? (Optional)
- N/A

**Comment by jeet-dhandha** [comment]
### Before Fix:

https://github.com/Expensify/App/assets/78416198/a093565d-a599-4a8f-9a56-9bc439f20bf0


### After Fix (Including above mentioned PR changes):


https://github.com/Expensify/App/assets/78416198/4a40f873-94d4-4886-86b5-ff573d0bae5b

**Comment by alexpensify** [comment]
@parasharrajat - are you suggesting that we should move this to be on HOLD until that GH is resolved?

**Comment by jeet-dhandha** [comment]
@parasharrajat My solution will work in current `main` branch but if we merge the mentioned PR, then the issue will rise again and will then be counted as regression. So to avoid that i mentioned it here.

**Comment by thienlnam** [comment]
I think we should hold all these for https://github.com/Expensify/App/issues/26239 which should ideally fix the root cause for all these focus issues upstream

**Comment by alexpensify** [comment]
Cool, moving this to weekly for now

**Comment by jeet-dhandha** [comment]
- Mobile Safari currently focuses on Input elements when their selection is set to greater than 0, which we don't want to happen.
- My solution basically uses the hack we have been using for Mobile Safari where we set Selection to `{start: 0, end: 0}` to not trigger the `:focus` of input.
- We are fixing this upstream here,
- https://github.com/WebKit/WebKit/pull/15092
- But till then user's will still need this workaround, as upstream PR might take longer.

Also https://github.com/Expensify/App/pull/28790 this PR is merged so we can move on with this issue.

**Comment by thienlnam** [comment]
I wouldn't consider this a severe enough bug to need to get a fix out immediately - let's just wait for the upstream fix so we don't have to have a bunch of back and fourths

**Comment by alexpensify** [comment]
Weekly update: On Hold

**Comment by alexpensify** [comment]
Weekly update: On Hold and will probably be on hold for an extended period of time

**Comment by alexpensify** [comment]
Weekly update: On Hold

**Comment by thienlnam** [comment]
Throwing this on a monthly since we're waiting on the new change to be included in the new version release of Safari

**Comment by alexpensify** [comment]
Thanks for that update @thienlnam!

**Comment by alexpensify** [comment]
Monthly Update: I think there is a new version of Safari now, but not sure if that's the one we are waiting for. I'll set a reminder to check in early January.

**Comment by alexpensify** [comment]
@thienlnam - is this one waiting on 17.3 too? Similar to the other GH: https://github.com/Expensify/App/issues/26239

**Comment by alexpensify** [comment]
Monthly Update: #26239 is waiting on Safari 17.4 to go into production. We are still at 17.3. Setting a reminder to check in next month.

**Comment by alexpensify** [comment]
Monthly Update: We are at 17.3.1. Setting a reminder to check in next month.

**Comment by alexpensify** [comment]
@ntdiary - I believe that we are at Version 17.4.1 now, can we confirm if we can take this one off hold? Thanks!

**Comment by alexpensify** [comment]
We are back open for proposals here. Thanks!

**Comment by alexpensify** [comment]
I'm going to put this back on hold for #26239. It looks like there will be some cleanup in the other GH.

**Comment by alexpensify** [comment]
Weekly Update: The other GH is going through a clean-up process.

**Comment by alexpensify** [comment]
Weekly Update: The other GH is still being updated

**Comment by alexpensify** [comment]
Weekly Update: The other GH is on hold for another update

**Comment by alexpensify** [comment]
Weekly Update: Same as last week, waiting on the other GH.

**Comment by alexpensify** [comment]
Weekly Update: https://github.com/Expensify/App/issues/26239 is moving along, we can test this one again once it's in Staging or production.

**Comment by alexpensify** [comment]
## Heads up, I will be offline until Tuesday, May 28, 2024, and will not actively watch over this GitHub during that period.

If anything urgent is needed here, please ask for help in the #expensify-open-source Slack Room-- thanks!

**Comment by alexpensify** [comment]
https://github.com/Expensify/App/issues/26239 is in production. @Expensify/applause can you please recheck this issue? Is it still valid based on the latest regression steps? Thanks! 



cc: @ntdiary who worked on the PR above.

**Comment by alexpensify** [comment]
Weekly Update: Waiting for a retest.

**Comment by alexpensify** [comment]
Weekly Update: I need to bump the Slack room, no movement here yet for the retest

**Comment by alexpensify** [comment]
Weekly Update: Waiting on a retest.

**Comment by m-natarajan** [comment]
Issue is not reproducible anymore.

https://github.com/Expensify/App/assets/38435837/f021f679-543d-4257-b580-014812df3645

**Comment by alexpensify** [comment]
Already did @parasharrajat. 😄 

Thank you everyone for the help here. I'm going to close since the other GH fixed this one!
